src/parser.cpp

```diff
@@ -458,17 +458,25 @@ void parser_t::job_add(shared_ptr<job_t> job) {
     job_list.insert(job_list.begin(), std::move(job));
 }
 
-void parser_t::job_promote(job_t *job) {
+void parser_t::job_promote(job_list_t::iterator job_it) {
+    // Move the job to the beginning.
+    std::rotate(job_list.begin(), job_it, std::next(job_it));
+}
+
+void parser_t::job_promote(const job_t *job) {
     job_list_t::iterator loc;
     for (loc = job_list.begin(); loc != job_list.end(); ++loc) {
         if (loc->get() == job) {
             break;
         }
     }
     assert(loc != job_list.end());
+    job_promote(loc);
+}
 
-    // Move the job to the beginning.
-    std::rotate(job_list.begin(), loc, std::next(loc));
+void parser_t::job_promote(size_t job_pos) {
```

Comment:
Can you rename this to `job_promote_at()` or similar? I don't like using the numbered cxx overloads and we're going to need to rename it when we port it to rust anyway (if it's still used at that point).